store/store.go

```diff
@@ -583,46 +701,64 @@ func (c *MongoStoreClient) GetLoopableMedtronicDirectUploadIdsAfter(userID strin
 		"deviceModel": bson.M{"$in": []string{"523", "523K", "554", "723", "723K", "754"}},
 	}
 
-	legacyObjects := []struct {
-		UploadID string `bson:"uploadId"`
-	}{}
-
-	objects := []struct {
+	type Object struct {
 		UploadID string `bson:"uploadId"`
-	}{}
+	}
+	var legacyObjects []Object
+	var objects []Object
 
 	cursor, err := dataCollection(c).Find(c.context, query, opts)
 	if err != nil {
 		return nil, err
 	}
 
-	legacyCursor, err := dataCollection(c).Find(c.context, legacyQuery, legacyOpts)
-	if err != nil {
-		return nil, err
+	collections := []*mongo.Collection{
+		dataCollection(c),
+		dataSetsCollection(c),
 	}
+	for _, collection := range collections {
+		if collection.Name() == dataSetsCollectionName {
+			delete(query, "type")
+		} else {
+			query["type"] = "upload"
```

Comment:
We don't need these if/else blocks at all - the query will always have `type`=`upload`, because the query is static.